File: 0445-add-two-numbers-ii/0445-add-two-numbers-ii.c
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2) {
    int s1[10000], s2[10000];
    int top1 = 0, top2 = 0;

    // Push values of l1 into stack
    while (l1) {
        s1[top1++] = l1->val;
        l1 = l1->next;
    }

    // Push values of l2 into stack
    while (l2) {
        s2[top2++] = l2->val;
        l2 = l2->next;
    }

    int carry = 0;
    struct ListNode* head = NULL;

    while (top1 > 0 || top2 > 0 || carry) {
        int sum = carry;

        if (top1 > 0) sum += s1[--top1];
        if (top2 > 0) sum += s2[--top2];

        struct ListNode* node = malloc(sizeof(struct ListNode));
        node->val = sum % 10;
        node->next = head;
        head = node;

        carry = sum / 10;
    }

    return head;
}

File: 0445-add-two-numbers-ii/0445-add-two-numbers-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
def create_linked_list(arr):
    if not arr:
        return None

    head = ListNode(int(arr[0]))
    current = head

    for val in arr[1:]:
        current.next = ListNode(int(val))
        current = current.next

    return head

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        s1, s2 = [], []
        while l1:
            s1.append(l1.val)
            l1 = l1.next
        while l2:
            s2.append(l2.val)
            l2 = l2.next
        
        ans=int("".join(map(str,s1)))+int("".join(map(str,s2)))
        ans=list(str(ans)) 
        return create_linked_list(ans)